osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -49,36 +72,48 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
 
             return new OsuDifficultyAttributes
             {
-                StarRating = starRating,
+                StarRating = sr,
                 Mods = mods,
-                AimStrain = aimRating,
-                SpeedStrain = speedRating,
+                Length = mapLength,
+
+                TapSr = tapSr,
+                TapDiff = tapDiff,
+                StreamNoteCount = streamNoteCount,
+                MashTapDiff = mashTapDiff,
+
+                FingerControlSr = fingerControlSr,
+                FingerControlDiff = fingerControlDiff,
+
+                AimSr = aimSr,
+                AimDiff = aimDiff,
+                AimHiddenFactor = aimHiddenFactor,
+                ComboTps = comboTps,
+                MissTps = missTps,
+                MissCounts = missCounts,
+                CheeseNoteCount = cheeseNoteCount,
+                CheeseLevels = cheeseLevels,
+                CheeseFactors = cheeseFactors,
+
                 ApproachRate = preempt > 1200 ? (1800 - preempt) / 120 : (1200 - preempt) / 150 + 5,
                 OverallDifficulty = (80 - hitWindowGreat) / 6,
-                MaxCombo = maxCombo,
-                Skills = skills
+                MaxCombo = maxCombo
             };
         }
 
-        protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(IBeatmap beatmap, double clockRate)
+        protected override Skill[] CreateSkills(IBeatmap beatmap)
         {
-            // The first jump is formed by the first two hitobjects of the map.
-            // If the map has less than two OsuHitObjects, the enumerator will not return anything.
-            for (int i = 1; i < beatmap.HitObjects.Count; i++)
-            {
-                var lastLast = i > 1 ? beatmap.HitObjects[i - 2] : null;
-                var last = beatmap.HitObjects[i - 1];
-                var current = beatmap.HitObjects[i];
+            throw new NotImplementedException();
+        }
 
-                yield return new OsuDifficultyHitObject(current, lastLast, last, clockRate);
-            }
+        protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(IBeatmap beatmap, double clockRate)
+        {
+            throw new NotImplementedException();
         }
 
-        protected override Skill[] CreateSkills(IBeatmap beatmap) => new Skill[]
+        protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate)
         {
-            new Aim(),
-            new Speed()
-        };
+            throw new NotImplementedException();
+        }
```

Comment:
What's the explanation for these losing their implementations?

Reply:
New system isn't using hitobject strain based skills so it can't really fit into current skill implementation